lib/semantics/check-declarations.cc

```diff
@@ -106,6 +107,41 @@ void CheckHelper::Check(const Symbol &symbol) {
   if (isAssociated) {
     return;  // only care about checking VOLATILE on associated symbols
   }
+  if (symbol.has<ProcBindingDetails>()) {
+    CheckBinding(symbol);
+    return;
+  }
+  if (const auto *details{symbol.detailsIf<DerivedTypeDetails>()}) {
+    CHECK(symbol.scope());
+    CHECK(symbol.scope()->symbol() == &symbol);
+    CHECK(symbol.scope()->IsDerivedType());
+    if (symbol.attrs().test(Attr::ABSTRACT) &&
+        (symbol.attrs().test(Attr::BIND_C) ||
+            (details && details->sequence()))) {
```

Comment:
Why not using `IsExtensibleType` here  also ?

Reply:
Because I don't have a `DerivedTypeSpec`.